config.go

```diff
@@ -0,0 +1,223 @@
+/*
+  Copyright 2011 Alec Thomas
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"io"
+	"os"
+	"os/user"
+	"strings"
+	"sync"
+)
+
+var (
+	instance *config
+	once     sync.Once
+)
+
+type config struct {
+	FGColors   map[Priority]string
+	BGColors   map[Priority]string
+	Priority   string
+	Graft      string
+	File       string
+	LegacyFile string
+	Order      string
+	ConfigFile string
+}
+
+// Map for priority level to ANSI colour
+var colourPriorityMap = map[Priority]string{
+	VERYHIGH: BRIGHT + FGRED,
+	HIGH:     BRIGHT + FGYELLOW,
+	MEDIUM:   FGWHITE,
+	LOW:      FGCYAN,
+	VERYLOW:  FGBLUE,
+}
+
+func GetConfigInstance() *config {
+	once.Do(func() {
+		fgColors := map[Priority]string{
+			VERYLOW:  BLUE,
+			LOW:      CYAN,
+			MEDIUM:   WHITE,
+			HIGH:     BRIGHTYELLOW,
+			VERYHIGH: BRIGHTRED,
+		}
+		bgColors := map[Priority]string{
+			VERYLOW:  NOCOLOR,
+			LOW:      NOCOLOR,
+			MEDIUM:   NOCOLOR,
+			HIGH:     NOCOLOR,
+			VERYHIGH: NOCOLOR,
+		}
+		priority := "medium"
+		graft := "root"
+		legacyFile := ".todo"
+		order := "priority"
+		currentUser, err := user.Current()
+		taskListFile := ".todo2"
+		configFile := ".todorc"
+		if err == nil {
+			configFile = strings.Join([]string{currentUser.HomeDir, configFile}, "/")
+		}
+		instance = &config{
+			BGColors:   bgColors,
+			FGColors:   fgColors,
+			Priority:   priority,
+			Graft:      graft,
+			File:       taskListFile,
+			LegacyFile: legacyFile,
+			Order:      order,
+			ConfigFile: configFile,
+		}
+	})
+	return instance
+}
+
+type ConfigIO interface {
+	Deserialize(reader io.Reader) (*MarshalableConfig, error)
+}
+
+type ConfigIOImpl struct {
+}
+
+func NewConfigIO() ConfigIO {
+	return ConfigIOImpl{}
+}
+
+type MarshalableConfig struct {
+	BGColors   map[string]string
+	FGColors   map[string]string
+	Priority   string
+	Graft      string
+	File       string
+	LegacyFile string
+	Order      string
+}
```

Comment:
You shouldn't need this. Use the previous `config` struct, but implement `json.Marshaler` and `json.Unmarshaler` for `Priority`.